Migration doc - What has changed Update (#2997)
wso2/docs-is: en/docs/deploy/migrate/what-has-changed.md

- The artifact ID of the authenticator has been updated.

> - The ```emailotpauthethentication``` endpoint which was packed with the product is now an integral part of the authentication portal. Following are the steps to integrate their changes to Identity Server 5.12.0 for users with customized email TOTP pages.

Reviewer: ```suggestion
- The ```emailotpauthethentication``` endpoint, which was packed with the product is now an integral part of the authentication portal. Following are the steps to integrate their changes to Identity Server 5.12.0 for users with customized email TOTP pages.
```

- It’s no longer required to add the TOTP authenticator into the ```dropins``` folder as the authenticator has been moved into the ```plugins``` directory by default in IS 5.12.0.

> - The ```totpauthethentication``` endpoint which was packed with the product is now an integral part of the authentication portal. Following are the steps to integrate their changes to Identity Server 5.12.0 for users with customized TOTP pages.

Reviewer: ```suggestion
- The ```totpauthethentication``` endpoint, which was packed with the product is now an integral part of the authentication portal. Following are the steps to integrate their changes to Identity Server 5.12.0 for users with customized TOTP pages.
```

        TOTPAuthenticationEndpointEnableTOTPPage="totpauthenticationendpoint/enableTOTP.jsp"
        ```

>     3. Unlike the earlier versions of the Identity server, the TOTP authenticator of IS 5.12.0 does not offer sending the TOTP via email as a default option. Add the following configuration to the ```deployment.toml``` file, to enable sending the TOTP through Email.

Reviewer: ```suggestion
    3. Unlike the earlier versions of the Identity server, the TOTP authenticator of IS 5.12.0 does not offer sending the TOTP via email as a default option. Add the following configuration to the ```deployment.toml``` file to enable sending the TOTP through Email.
```

```

## API Endpoints
This section covers the updates related to APIs and API endpoints on Identity Server 5.12.0.

### SCIM2 Endpoint

> This section covers the updates related to SCIM2 endpoint on Identity Server 5.12.0.

Reviewer: ```suggestion
This section covers the updates related to the SCIM2 endpoint on Identity Server 5.12.0.
```

To migrate the H2 databases to the newer versions, follow the instructions given below.
1. Download the [migration.sh script](https://github.com/wso2/samples-is/blob/master/h2-migration-tool/migration.sh) and run it using the command ```sh migration.sh``.

    Alternatively, you could run the following command to download and run the script in one go.

    ``` js
    curl -s https://raw.githubusercontent.com/wso2/samples-is/master/h2-migration-tool/migration.sh | bash
    ```

2. Once you run the script, you will be prompted with the following questions on the terminal:
    - **Enter the path to the previous database files:**
        If the database being migrated is the embedded H2 database, the path should be ```<OLD_IS_HOME>/repository/database```.
    - **Enter the path to store the newly created files:**
        Enter the path of the target directory where the new database files should be generated.

3. After successfully executing the migration script, copy the files in the target directory to the ```repository/database``` directory in the new Identity Server 5.12.0 pack for embedded H2 database migration.

## Data providers

> This section covers the updates related to data provider configurations on Identity Server 5.12.0.

Reviewer: ```suggestion
This section covers the updates related to data provider configurations on WSO2 Identity Server 5.12.0.
```

### Introspection Data Providers

> Up until Identity Server 5.11.0, the following single configuration was used to enable or disable all the registered OAuth introspection data providers.

Reviewer: ```suggestion
Up until WSO2 Identity Server 5.11.0, the following single configuration was used to enable or disable all the registered OAuth introspection data providers.
```

```

## System configs
This section covers the updates related to system configurations on Identity Server 5.12.0.

### Default CORS Configuration

> In previously versions of Identity Server, the default HTTP methods allowed for CORS were only ```GET```, ```POST```, ```HEAD``` and ```OPTIONS```.

Reviewer: ```suggestion
In previously versions of WSO2 Identity Server, the default HTTP methods allowed for CORS were only ```GET```, ```POST```, ```HEAD``` and ```OPTIONS```.
```

In Identity Server 5.12.0, this has been changed to allow the following HTTP methods ```GET```, ```POST```, ```PUT```, ```PATCH```, ```DELETE```, ```HEAD``` and ```OPTIONS```.

Learn more on [how to change the CORS configuration](https://is.docs.wso2.com/en/latest/learn/cors).

### Log4j2 logging in Hazelcast

> If you have been using Identity Server in a [Hazelcast cluster](https://is.docs.wso2.com/en/latest/administer/configuring-hazelcast/), you may have configured the logging type for Hazelcast as log4j ```(Log4j1)```. Log4j1 logging is no longer supported in WSO2 Identity Server.

Reviewer: ```suggestion
If you have been using WSO2 Identity Server in a [Hazelcast cluster](https://is.docs.wso2.com/en/latest/administer/configuring-hazelcast/), you may have configured the logging type for Hazelcast as log4j ```(Log4j1)```. Log4j1 logging is no longer supported in WSO2 Identity Server.
```

> In Identity Server 5.12.0, the Log4j version is upgraded to ```Log4j2```. Therefore the Hazelcast configuration needs to be updated to ```log4j2``` by adding the following configuration to the ```deployment.toml``` file.

Reviewer: ```suggestion
In WSO2 Identity Server 5.12.0, the Log4j version is upgraded to ```Log4j2```. Therefore the Hazelcast configuration needs to be updated to ```log4j2``` by adding the following configuration to the ```deployment.toml``` file.
```
